books/kestrel/c/syntax/validation-information.lisp

```diff
@@ -78,6 +78,55 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
+(fty::defprod uid
+  :short "Fixtype of unique identifiers."
+  :long
+  (xdoc::topstring
+   (xdoc::p
+    "These are numerical identifiers which are intended
+     to be unique to a given variable, function, type name, etc.
+     E.g., there may be many variables throughout a program
+     with the name @('x'), but all such distinct variables
+     will have distinct unique identifiers.")
+   (xdoc::p
+    "Unique identifiers are assigned during validation
+     to aid subsequent analysis.
+     By annotating identifiers with their unique alias,
+     disambiguation of variables becomes trivial."))
+  ((ident nat))
```

Comment:
Maybe I would call this component `uid` instead of `ident`, since it's not a C identifiers as such.

(It's often awkward to find a good name for a one-component product type. Sometimes I've used `unwrap` but I've been moving away from that.)